pymiele/pymiele.py

```diff
@@ -158,11 +159,11 @@ async def listen_events(
                         if event_type == "event: devices":
                             data = json.loads(data_line[6:])
                             if data_callback is not None:
-                                asyncio.create_task(data_callback(data))
+                                asyncio.create_task(data_callback(data))  # noqa: RUF006
                         elif event_type == "event: actions":
                             data = json.loads(data_line[6:])
                             if actions_callback is not None:
-                                asyncio.create_task(actions_callback(data))
+                                asyncio.create_task(actions_callback(data))  # noqa: RUF006
```

Comment:
This should be fixed in a future PR. See https://docs.astral.sh/ruff/rules/asyncio-dangling-task/

Or maybe avoid using awaitables as callbacks and let the calling/consuming code wrap that in a coroutine if needed (which can also save some unnecessary waiting for loop iterations).

Reply:
I remember that it took a lot of work to get the listener/callback/restart/ structure going. It is stable now from a practical viewpoint. But I have no problems with trying to improve the code. We can always fall back to present version if needed.